core/web/api/links.py

```diff
@@ -66,3 +66,51 @@ def multiupdate(self):
             updated.append(link.id)
 
         return render({"updated": updated})
+
+    @route("/", methods=['POST'])
+    @requires_permissions('write')
+    def new(self):
+        """Create a new link
+
+        Create a new link from the JSON object passed in the ``POST`` data.
+
+        :<json object params: JSON object containing object ids to link
+        """
+
+        type_map = {
+            'observable': observables.Observable,
+            'entity': entities.Entity,
+            'indicator': indicators.Indicator
+        }
+
+        mandatory_params = ['type_src', 'type_dst', 'link_src', 'link_dst']
+        params = request.json
+
+        if all(key in params for key in mandatory_params):
+
+            type_src = params.get('type_src')
+            type_dst = params.get('type_dst')
+            src_object_class = type_map.get(type_src, None)
+            dst_object_class = type_map.get(type_dst, None)
+
+            if src_object_class and dst_object_class:
+                src = get_object_or_404(src_object_class, id=params.get("link_src"))
+                dst = get_object_or_404(dst_object_class, id=params.get("link_dst"))
+
+                if params.get("first_seen", None) and params.get("last_seen", None):
+                    link = src.link_to(dst,
+                                       params.get("description",None),
+                                       params.get("source", None),
+                                       params.get("first_seen"),
```

Comment:
Same comment here, it's unclear which keys you're OK with having and which not. Keys that *must* be included in the dictionary should be accessed through `params[key]` rather than `params.get(key)` (and ideally their presence should be checked, just like you do in L89).

Optional keys can be accessed with `.get`, so that you can easily provide a default value if they aren't present.